app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt

```diff
@@ -69,6 +70,11 @@ class SearchDialogController(
 ) : SearchController {
 
     override fun handleUrlCommitted(url: String, fromHomeScreen: Boolean) {
+        // Do not load URL if search engine is application added
+        if (fragmentStore.state.searchEngineSource.searchEngine?.type == SearchEngine.Type.APPLICATION) {
+            return
```

Comment:
That's one of the requirements.  History should not trigger a URL load.  The user only searches history and either tap on the result or back out.  Maybe we need some kind of "no result" line?

Reply:
Just pointing it out since I was just investigating what the code did. I think we should just file a followup issue to have UX investigate what if anything we should do in such a state. No changes needed, but felt it was noteworthy.

Reply:
Agreed, issue created to track this https://github.com/mozilla-mobile/fenix/issues/25282